src/pytest_inmanta_lsm/load_generator.py

```diff
@@ -0,0 +1,199 @@
+            del self._target, self._args, self._kwargs
+
+    def join(self, timeout=None):
+        """Wait until the thread terminates.
+
+        This blocks the calling thread until the thread whose join() method is
+        called terminates -- either normally or through an unhandled exception
+        or until the optional timeout occurs.
+
+        We override it to raise any exception that the Thread could have faced
+        """
+        super().join(timeout)
+        # we re-raise the caught exception
+        # if any was caught
+        if self.exc:
+            raise self.exc
+
+
+class LoadGenerator:
+    def __init__(
+        self,
+        remote_orchestrator: remote_orchestrator.RemoteOrchestrator,
+        service_type: str,
+        logger: logging.Logger = LOGGER,
+        sleep_time: float = 1.0,
+    ):
+        self.remote_orchestrator = remote_orchestrator
+        self.service_type = service_type
+        self.sleep_time = sleep_time
+        self.logger = logger
+        self._lock = threading.Lock()
+        self.SHOULD_THREAD_RUN = True
+        self._thread = LoadThread(target=self.between_callback, daemon=True, name="Thread-LG")
+
+    def __enter__(self) -> None:
+        self.logger.debug("Starting Thread %s", self._thread.name)
+        self._thread.start()
+
+    def __exit__(self, exc_type, exc_value, exc_traceback):
+        self.change_status()
+        self.logger.debug("Stopping Thread %s", self._thread.name)
+        self._thread.join(30)
+        self.logger.debug("Thread %s has been stopped, closing session", self._thread.name)
+
+    def fetch_thread_status(self) -> bool:
+        """
+        Retrieve the status of whether the thread should keep running or not
+        """
+        with self._lock:
+            return self.SHOULD_THREAD_RUN
+
+    def change_status(self):
+        """
+        Change the `SHOULD_THREAD_RUN` status
+        """
+        with self._lock:
+            self.SHOULD_THREAD_RUN = False
+        self.logger.debug("Thread %s should stop", self._thread.name)
+
+    def between_callback(self):
+        loop = asyncio.new_event_loop()
+        asyncio.set_event_loop(loop)
+
+        loop.run_until_complete(self.create_load())
+        loop.close()
+
+    async def create_load(self):
+        """
+        Loop to run provided tasks. This loop will only stop when `SHOULD_THREAD_RUN` is set to False
+        """
+        start_datetime = datetime.datetime.utcnow()
+        nb_datapoints = 15
+        # start_interval and end_interval should be at least <nb_datapoints> minutes separated from each other
+        # But we need also to respect the following constraint:
+        # When round_timestamps is set to True, the number of hours between start_interval and end_interval should be
+        # at least the amount of hours equal to nb_datapoints
+        end_datetime = start_datetime + datetime.timedelta(hours=nb_datapoints + 1)
+
+        while True:
```

Comment:
I would spread the load more evenly over the interval: now you are machine-gunning the api at the start of the interval.